documentation/api/io.strimzi.api.kafka.model.listener.arraylistener.GenericKafkaListener.adoc

```diff
@@ -0,0 +1,323 @@
+Configures listeners to connect to Kafka brokers within and outside Kubernetes.
+
+You configure the listeners in the `Kafka` resource.
+
+.Example `Kafka` resource showing listener configuration
+[source,yaml,subs="+attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    #...
+    listeners:
+      - name: plain
+        port: 9092
+        type: internal
+        tls: false
+      - name: tls
+        port: 9093
+        type: internal
+        tls: true
+        authentication:
+          type: tls
+      - name: external1
+        port: 9094
+        type: route
+        tls: true
+      - name: external2
+        port: 9095
+        type: ingress
+        tls: false
+        authentication:
+          type: tls
+        configuration:
+          bootstrap:
+            host: bootstrap.myingress.com
+          brokers:
+          - broker: 0
+            host: broker-0.myingress.com
+          - broker: 1
+            host: broker-1.myingress.com
+          - broker: 2
+            host: broker-2.myingress.com
+    #...
+----
+
+[id='property-generic-listeners-{context}']
+=== `listeners`
+You configure Kafka broker listeners using the `listeners` property in the `Kafka` resource.
+Each listener is defined as an array.
+
+.Example listener configuration
+[source,yaml,subs="+attributes"]
+----
+listeners:
+  - name: plain
+    port: 9092
+    type: internal
+    tls: false
+----
+
+The name and port must be unique within the Kafka cluster.
+The name can be up to 25 characters long, comprising lower-case letters and numbers.
+
+By specifying a unique name and port for each listener,
+you can configure multiple listeners.
+
+=== `type`
+
+The type is set as `internal`,
+or for external listeners, as `route`, `loadbalancer`, `nodeport` or `ingress`.
+
+internal:: You can configure internal listeners with or without encryption using the `tls` property.
++
+.Example `internal` listener configuration
+[source,yaml,subs="+attributes"]
+----
+#...
+spec:
+  kafka:
+    #...
+    listeners:
+      #...
+      - name: plain
+        port: 9092
+        type: internal
+        tls: false
+      - name: tls
+        port: 9093
+        type: internal
+        tls: true
+        authentication:
+          type: tls
+    #...
+----
+route:: Configures an external listener to expose Kafka using OpenShift `Routes` and the HAProxy router.
++
+A dedicated `Route` is created for every Kafka broker pod.
+An additional `Route` is created to serve as a Kafka bootstrap address.
+Kafka clients can use these `Routes` to connect to Kafka on port 443.
```

Comment:
Should we highlight more this? I mean, the user could be confused by the fact that we mention port 443 but the example has 9094. So explaining that TLS on routes is enabled by default and it works on port 443 but the traffic is routed to the port 9094 configured in the example?

Reply:
Changed to:

> The client connects on port 443, the default router port, but traffic is then routed to the port you configure,
> which is `9094` in this example.

Reply:
Did the same for ingress:

> The client connects on port 443, the default controller port, but traffic is then routed to the port you configure,
> which is `9095` in this example.